connect/runtime/src/main/java/org/apache/kafka/connect/runtime/health/ConnectClusterStateImpl.java

```diff
@@ -42,13 +45,22 @@ public ConnectClusterStateImpl(HerderProvider herderProvider) {
 
     @Override
     public Collection<String> connectors() {
+        final CountDownLatch connectorsLatch = new CountDownLatch(1);
         final Collection<String> connectors = new ArrayList<>();
         herderProvider.get().connectors(new Callback<java.util.Collection<String>>() {
             @Override
             public void onCompletion(Throwable error, Collection<String> result) {
                 connectors.addAll(result);
+                connectorsLatch.countDown();
             }
         });
+        try {
+            // Should take no more than 10 seconds to get a list of connectors. Any longer than this
+            // and it's very likely something is wrong.
+            connectorsLatch.await(10, TimeUnit.SECONDS);
```

Comment:
nit: should we log an error if this method returns false?

Reply:
Hmm... good point. I think it may make more sense raise a `ConnectException` in order to avoid incorrect/unexpected behavior from the code that's invoking this method. WDYT? Would be interested in @mageshn's thoughts as well

Reply:
Yes, it would probably be good to throw an exception here if the method returns false.